crates/ruff_linter/src/rules/flake8_use_pathlib/rules/invalid_pathlib_with_suffix.rs

```diff
@@ -44,21 +49,32 @@ use ruff_text_size::Ranged;
 /// for a given situation (it's possible that the string was correct
 /// but was being passed to the wrong method entirely, for example).
 #[derive(ViolationMetadata)]
-pub(crate) struct DotlessPathlibWithSuffix;
+pub(crate) struct InvalidPathlibWithSuffix {
+    // TODO: Since "." is a correct suffix in Python 3.14,
+    // the rule should revert to its original behavior.
+    single_dot: bool,
+}
+
+impl Violation for InvalidPathlibWithSuffix {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
 
-impl AlwaysFixableViolation for DotlessPathlibWithSuffix {
     #[derive_message_formats]
     fn message(&self) -> String {
-        "Dotless suffix passed to `.with_suffix()`".to_string()
+        "Invalid suffix passed to `.with_suffix()`".to_string()
```

Comment:
The original message is more helpful in the majority of cases.

```suggestion
        if !self.single_dot {"Dotless suffix passed to `.with_suffix()`".to_string()} else {
            "Invalid suffix passed to `.with_suffix()`".to_string()
         }
```